### PR TITLE
feat: withdraw and swap adapter

### PR DESCRIPTION
**Pull request by grothem**
- Adds in a new adapter that allows a user to withdraw an asset and swap it to a different asset. This is based on the existing liquidity swap adapter, but instead of depositing the swapped assets back into the pool, they are transferred to the user.

### CODE REVIEW

**Comment by grothem** [comment]
Link T-13768

File: contracts/adapters/paraswap/ParaSwapWithdrawSwapAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  constructor(
+    IPoolAddressesProvider addressesProvider,
+    IParaSwapAugustusRegistry augustusRegistry,
+    address owner
+  ) BaseParaSwapSellAdapter(addressesProvider, augustusRegistry) {
+    transferOwnership(owner);
+  }
+
+  function executeOperation(
```
Comment (miguelmtzinf) [inline contracts/adapters/paraswap/ParaSwapWithdrawSwapAdapter.sol:26]: remove the entire function

File: contracts/adapters/paraswap/ParaSwapWithdrawSwapAdapter.sol [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import {IPoolAddressesProvider} from '@aave/core-v3/contracts/interfaces/IPoolAddressesProvider.sol';
+import {BaseParaSwapSellAdapter} from './BaseParaSwapSellAdapter.sol';
+import {IParaSwapAugustusRegistry} from './interfaces/IParaSwapAugustusRegistry.sol';
+import {SafeERC20} from '@aave/core-v3/contracts/dependencies/openzeppelin/contracts/SafeERC20.sol';
+import {SafeMath} from '@aave/core-v3/contracts/dependencies/openzeppelin/contracts/SafeMath.sol';
+import {IParaSwapAugustus} from './interfaces/IParaSwapAugustus.sol';
+import {ReentrancyGuard} from '../../dependencies/openzeppelin/ReentrancyGuard.sol';
+
+contract ParaSwapWithdrawSwapAdapter is BaseParaSwapSellAdapter, ReentrancyGuard {
+  using SafeMath for uint256;
```
Comment (miguelmtzinf) [inline contracts/adapters/paraswap/ParaSwapWithdrawSwapAdapter.sol:15]: remove

File: contracts/adapters/paraswap/ParaSwapWithdrawSwapAdapter.sol [hunk 3]
```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.10;
+
+import {IERC20Detailed} from '@aave/core-v3/contracts/dependencies/openzeppelin/contracts/IERC20Detailed.sol';
+import {IERC20WithPermit} from '@aave/core-v3/contracts/interfaces/IERC20WithPermit.sol';
+import {IPoolAddressesProvider} from '@aave/core-v3/contracts/interfaces/IPoolAddressesProvider.sol';
+import {BaseParaSwapSellAdapter} from './BaseParaSwapSellAdapter.sol';
+import {IParaSwapAugustusRegistry} from './interfaces/IParaSwapAugustusRegistry.sol';
+import {SafeERC20} from '@aave/core-v3/contracts/dependencies/openzeppelin/contracts/SafeERC20.sol';
+import {SafeMath} from '@aave/core-v3/contracts/dependencies/openzeppelin/contracts/SafeMath.sol';
```
Comment (miguelmtzinf) [inline contracts/adapters/paraswap/ParaSwapWithdrawSwapAdapter.sol:10]: remoev

Comment (grothem) [reply contracts/adapters/paraswap/ParaSwapWithdrawSwapAdapter.sol]: done

**Review by miguelmtzinf — APPROVED** [review]

**Pull request merged**